### ISSUE TITLE
Integrating with GRAPE

### ISSUE THREAD

**Issue by LucaCappelletti94** [biopragmatics/bioregistry#498]
Hello, and thank you for creating this resource.

I wondered whether we could spend a bit of time to plug this (optionally) in GRAPE's graph retrieval to uniform the node names of all the biological graphs therein upon user request, such as the OBO graphs.

The main thing that worries me is that I'm not sure how well this system scales with graphs with a few million nodes.

- Do you have some speed benchmarks?
- Is the system already parallelized?
- Is it Web-API based, or does it work offline?

Best,
Luca

**Comment by LucaCappelletti94** [comment]
We have integrated within the graph retrieval portion of GRAPE an option to normalize the graph using `bioregistry`. This 
should shortly be available from PyPI.

A couple of essential notes:
- So far, we have had to use Python multiprocessing to normalize the node names. This is not efficient and will be prone to failure in some use cases.
- We cannot integrate this directly in Rust, as we initially hoped, unless we ship with the library also a Python interpreter compiled with the library. This is, of course, unfeasible.
- The only solution we see to have bioregistry integrated with the faster Rust part of the GRAPE library is to write it in Rust, but we understand the amount of work that went into making this tool in Python forbids such an undertaking.

I will be posting here when the new version becomes available.
